### ISSUE TITLE
Something wrong with the hevals()

### ISSUE THREAD

**Issue by DevelopDaily** [softwareQinc/qpp#95]
Here is a test case.

```
    auto a = gt.T;
    std::cout<<disp(a);
    std::cout<<"\n--------------------------------\n\n";

    std::cout<<disp(hevals(a));
    std::cout<<"\n--------------------------------\n\n";

    std::cout<<disp(evals(a));
    std::cout<<"\n--------------------------------\n\n";

```
The same matrix produces two different sets of eigenvalues:

```
1                     0 
0   0.707107 + 0.707107i
--------------------------------

0.707107 
       1 
--------------------------------

                  1 
0.707107 + 0.707107i
--------------------------------

```
The `evals()` is correct and `hevals()` is not. The `T` gate is not even Hermitian, is it? Should `hevals()` throw an exception or something when its input is not Hermitian?

The problem seems to boil down to the the `Eigen::SelfAdjointEigenSolver<cmat>`, which does not check whether the input is Hermitian or not.

**Comment by vsoftco** [comment]
@DevelopDaily qpp doesn't do many of the more "high-level" (logic) checks, such as unitarity, hermiticity etc. It only throws exceptions for blatant mismatches, such as incorrect set of dimensions, tracing out non-existing subsystems etc. I decided to go this way since otherwise it'll be way too many "logic" checks that may significantly slow down the program. I've updated to Doxygen comments for each function so it makes it clear.
